js/nodes/solver_settings/structural_mechanic_solver.js

```diff
@@ -0,0 +1,54 @@
+class StructuralMechanicSolver {
+    constructor() {
+        this.addInput("model_import_settings", "map");      // 0
+        this.addInput("material_import_settings", "map");   // 1
+        this.addOutput("solver_settings", "map");
+        this.properties = {
+            "solver_type"                        : "Static",
+            "model_part_name"                    : "Structure",
+            "domain_size"                        : 2,
+            "echo_level"                         : 0,
+            "analysis_type"                      : "non_linear",
+            "model_import_settings"              : {
+                "input_type"     : "mdpa",
+                "input_filename" : "Ring"
+            },
+            "material_import_settings"           : {
+                "materials_filename" : "StructuralMaterials.json"
+            },
+            "time_stepping"                      : {
+                "time_step" : 1
+            },
+            "line_search"                        : false,
+            "convergence_criterion"              : "residual_criterion",
+            "displacement_relative_tolerance"    : 0.0001,
+            "displacement_absolute_tolerance"    : 1e-9,
+            "residual_relative_tolerance"        : 0.0001,
+            "residual_absolute_tolerance"        : 1e-9,
+            "max_iteration"                      : 10,
+            "rotation_dofs"                      : false,
```

Comment:
this should be variable, can you expose it like the domain size?

Reply:
ok, you know better :)

Reply:
In the other GUIs this is automatically switched on depending on the user-defined elements. In this case, automating it would imply parsing the mpda (which we already do) and go through all the element types in the elements section. Not sure if this will be possible.

What is not clear to me is how we can select this field from the properties (e.g. the properties of a shell element are almost identical to that of a 2D plane stress case without rotations).

Reply:
This is similar as the CL problem that appeared above. As @rubenzorrilla says we need to extract more info from the mpda, which is not a problem but also need to dynamically generate nodes based on inputs, which is something we are missing at the moment (but possible).

I would leave it as is for this PR and make a separate one to implement those changes.